### ISSUE TITLE
`seccomp_desched` test fails on older Linux kernels

### ISSUE THREAD

**Issue by rocallahan** [mozilla/rr#2047]
The test added here https://github.com/mozilla/rr/commit/ad057a32cf1a9afd6b6959a805a20d8623bf6695 is failing on `ci.rr-project.org`.

That machine is still running a 4.4 kernel. On that kernel, SECCOMP events precede PTRACE_SYSCALL events. The test triggers a fake "spurious" DESCHED signal just before a `read` call that the tracee is use a seccomp filter to turn into `SIGSYS`. The DESCHED signal arrives OK and causes rr to enter `handle_desched_event`. That code resumes execution with `RESUME_SYSCALL`, looking for the ptrace syscall event indicating we entered the maybe-blocking buffered syscall. However, before that happens, we get the SIGSYS SECCOMP event. We were not expecting a SECCOMP event here and assertions fire.

So basically, on <= 4.4 kernels, for tracees that are using SECCOMP filters to modify maybe-blocking buffered syscalls, spurious DESCHED events will cause rr to die.

**Comment by rocallahan** [comment]
For newer kernels there is no problem because we get the ptrace syscall event first, which causes `handle_desched_event` to set everything up and exit. Then later we receive the SECCOMP event and process that to turn the in-progress syscall into a SIGSYS or whatever else is required.

Fixing this for older kernels is tricky. We can detect the SECCOMP event and use that as the cue to complete `handle_desched_event`. However,  we still have to process the SECCOMP event somewhere as well.
